quarkus starts database testcontainer even if quarkus.datasource.start-container set to false

### Describe the bug

There is testcontainers lib in classpath
`quarkus.datasource.start-container` set to false in the application.properties
when run tests without any tests that starts testcontainer and Docker available it fails because quarkus tries to run testcontainer.  

It's needed to completely decouple unit tests and integration tests. (So it can be achieved by excluding integration test by JUnit 5 tag but it fails if run unit tests on a machine without Docker due to the reason described above).

### Expected behavior

`mvn clean test` should pass if it runs  only tests that doesn't use testcontainers and Docker is turned off.
when `quarkus.datasource.start-container` set to false 


### Actual behavior

It fails:
```
Caused by: io.quarkus.builder.BuildException: 
Build failure: Build failed due to errors
        [error]: Build step io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor#launchDatabases threw an exception: java.lang.IllegalStateException: Could not f
ind a valid Docker environment. Please see logs and check configuration
        at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$7(DockerClientProviderStrategy.java:215)
        at java.base/java.util.Optional.orElseThrow(Optional.java:408)
        at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:207)
        at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:136)
        at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:178)
        at org.testcontainers.LazyDockerClient.getDockerClient(LazyDockerClient.java:14)
        at org.testcontainers.LazyDockerClient.authConfig(LazyDockerClient.java:12)
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:310)
        at io.quarkus.devservices.postgresql.deployment.PostgresqlDevServicesProcessor$1.startDatabase(PostgresqlDevServicesProcessor.java:43)
        at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:217)
        at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:103)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```
As you can see there is call of 
`io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:103)`



### How to Reproduce?

Quarkus version:
- Create quarkus project add testcontainers, Postgres or download the existing simple demo project 
 [todo-list.zip](https://github.com/quarkusio/quarkus/files/6945398/todo-list.zip)
- Turn off Docker
- run `mvn clean test` in this project  (it's not going to run tests that uses testcontainer as they are excluded by surefire plugin with junit tags) 

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19042.1110]

### Output of `java -version`

java version "11.0.11" 2021-04-20 LTS Java(TM) SE Runtime Environment 18.9 (build 11.0.11+9-LTS-194) Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.11+9-LTS-194, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

1.13.7.Final (it's also reproducible for 2.1.1.Final but output is different)

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f) Maven home: C:\ProgramData\chocolatey\lib\maven\apache-maven-3.6.3\bin\.. Java version: 11.0.11, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-11.0.11 Default locale: en_US, platform encoding: Cp1251 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourceRuntimeConfig.java
+++ b/extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourceRuntimeConfig.java
@@ -36,13 +36,4 @@ public class DataSourceRuntimeConfig {
      */
     @ConfigItem
     public Optional<String> credentialsProviderName = Optional.empty();
-
-    /**
-     * If this is true then when running in dev or test mode Quarkus will attempt to start a testcontainers based
-     * database with these provided settings.
-     *
-     * This is not supported for all databases, and will not work in production.
-     */
-    @ConfigItem(defaultValue = "false")
-    public boolean startContainer;
 }

